### PR TITLE
feat(client): Planning Page

### PR DESCRIPTION
**Pull request by tiffzeng**
#

## Changes this PR introduces

- Planning Page (with mock data)
- Features two graphs (refer to figma)
- Refactored the way I was getting data in Orders page (from previous sprint)

## ClickUp reference 
nq0a3v

## Checklist

- [x] I renamed the title of my pull request to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] I have added tests to cover my changes (if applicable).

### CODE REVIEW

**Comment by Mackbellemore** [comment]
Task linked: [CU-nq0a3v Planning page](https://app.clickup.com/t/nq0a3v)

File: client/src/components/Planning/PlanningData.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+//Mock data for Planning Graphs
```
Comment (gordonpn) [inline client/src/components/Planning/PlanningData.ts]: This file doesn't need to be a TypeScript file as the frontend project is not a TypeScript project

File: client/src/components/views/Planning.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import React from 'react';
+import {
+  Bar,
+  BarChart,
+  CartesianGrid,
+  Legend,
+  Tooltip,
+  XAxis,
+  YAxis,
+  // eslint-disable-next-line import/no-unresolved
```
Comment (gordonpn) [inline client/src/components/views/Planning.jsx]: I think this line can be removed, if you have errors from ESLint in your IDE about this, try running `npm i` inside the client folder (not in the container).

File: client/src/constants.js [hunk 3]
```diff
@@ -136,6 +137,15 @@ export const appRoutes = [
...
     active: false,
   },
+  {
+    name: 'Planning',
+    protected: true,
+    allowedRoles: rolesAvailable,
+    path: '/planning',
+    component: lazy(() => import('components/views/Planning')),
+    icon: MdInsertChart,
+    active: false,
```
Comment (gordonpn) [inline client/src/constants.js]: I believe Sujan removed this property `active` in the PR that just merged, so just update the branch and remove this line and should be good!

File: client/src/constants.js [hunk 4]
```diff
@@ -136,6 +137,15 @@ export const appRoutes = [
     icon: MdGavel,
     active: false,
   },
+  {
+    name: 'Planning',
+    protected: true,
+    allowedRoles: rolesAvailable,
```
Comment (gordonpn) [inline client/src/constants.js]: I believe only Finance and Admin should have access to the planning page

File: client/src/constants.js (hunk 3 above)
Comment (tiffzeng) [reply client/src/constants.js]: sounds good!

File: client/src/constants.js (hunk 4 above)
Comment (tiffzeng) [reply client/src/constants.js]: oh okay, yeah it makes sense

**Review by gordonpn — DISMISSED** [review]

**Comment by DavidLiang01** [comment]
I think that you should forecast production for the upcoming years (2021, 2022, ....). What do you think?

In the Slack FAQ section:
```
Q: Minimum Requirements: Planning
A: Plan the production in the long term. Ex.: this year we will produce 100 bikes, the cost will be X and the profit Y.
```

**Comment by tiffzeng** [comment]
Sounds right, I'll adjust the data!

**Review by gordonpn — APPROVED** [review]

**Review by Mackbellemore — APPROVED** [review]

**Pull request merged**